test/test_ast_generation.py

```diff
@@ -8,11 +8,122 @@
 from pegen.grammar_parser import GeneratedParser as GrammarParser
 from pegen.testutil import parse_string, generate_parser_c_extension
 
-TEST_DIR = os.path.join("test", "test_data")
-PYTHON_SOURCE_FILENAMES = sorted(
-    filename for filename in os.listdir(TEST_DIR) if filename.endswith(".py")
-)
+TEST_CASES = [
+    ('annotated_assignment', 'x: int = 42\n'),
+    ('annotated_assignment_with_parens', '(paren): int = 3+2\n'),
+    ('annotated_assignment_with_yield', 'x: int = yield 42\n'),
+    ('annotated_no_assignment', 'x: int\n'),
+    ('annotation_with_parens', '(parens): int\n'),
+    ('assert', 'assert a\n'),
+    ('assert_message', 'assert a, b\n'),
+    ('asyncfor', 'async for i in a:\n    pass\n'),
```

Comment:
For multi-line files I'd like to see the structure of the file more clearly here, e.g. using a triple-quoted string (with some utility to strip indents) or just string concatenation, e.g.
```
    ('asyncfor',
        'async for i in a:\n'
        '    pass\n'),
```
(Or perhaps I'd like it even better if the \n wasn't present -- then you could make it a list of lines.)

Reply:
In commit 978bdf7 I have switched to multi-line strings (it reads better IMHO) and I have added preprocessing of the test cases that de-dents the string cases and also allows the possibility of using lists of strings as test cases (in which case they will be joined by newlines).